source/DataElement/AMDSFlatArray.cpp

```diff
@@ -0,0 +1,199 @@
+		}
+		break;
+	}
+	case AMDSDataTypeDefinitions::Unsigned16: {
+		QVector<quint16> *quint16SourceDataVector = &dataFlatArray.vectorQuint16();
+		QVector<quint16> *quint16TargetDataVector = &newDataFlatArray.vectorQuint16();
+		for (int i = 0; i < quint16TargetDataVector->size(); i++) {
+			quint16TargetDataVector->replace(i, quint16TargetDataVector->value(i) + quint16SourceDataVector->value(i));
+		}
+		break;
+	}
+	case AMDSDataTypeDefinitions::Signed32: {
+		QVector<qint32> *qint32SourceDataVector = &dataFlatArray.vectorQint32();
+		QVector<qint32> *qint32TargetDataVector = &newDataFlatArray.vectorQint32();
+		for (int i = 0; i < qint32TargetDataVector->size(); i++) {
+			qint32TargetDataVector->replace(i, qint32TargetDataVector->value(i) + qint32SourceDataVector->value(i));
+		}
+		break;
+	}
+	case AMDSDataTypeDefinitions::Unsigned32: {
+		QVector<quint32> *quint32SourceDataVector = &dataFlatArray.vectorQuint32();
+		QVector<quint32> *quint32TargetDataVector = &newDataFlatArray.vectorQuint32();
+		for (int i = 0; i < quint32TargetDataVector->size(); i++) {
+			quint32TargetDataVector->replace(i, quint32TargetDataVector->value(i) + quint32SourceDataVector->value(i));
+		}
+		break;
+	}
+	case AMDSDataTypeDefinitions::Signed64: {
+		QVector<qint64> *qint64SourceDataVector = &dataFlatArray.vectorQint64();
+		QVector<qint64> *qint64TargetDataVector = &newDataFlatArray.vectorQint64();
+		for (int i = 0; i < qint64TargetDataVector->size(); i++) {
+			qint64TargetDataVector->replace(i, qint64TargetDataVector->value(i) + qint64SourceDataVector->value(i));
+		}
+		break;
+	}
+	case AMDSDataTypeDefinitions::Unsigned64: {
+		QVector<quint64> *quint64SourceDataVector = &dataFlatArray.vectorQuint64();
+		QVector<quint64> *quint64TargetDataVector = &newDataFlatArray.vectorQuint64();
+		for (int i = 0; i < quint64TargetDataVector->size(); i++) {
+			quint64TargetDataVector->replace(i, quint64TargetDataVector->value(i) + quint64SourceDataVector->value(i));
+		}
+		break;
+	}
+	case AMDSDataTypeDefinitions::Float: {
+		QVector<float> *floatSourceDataVector = &dataFlatArray.vectorFloat();
+		QVector<float> *floatTargetDataVector = &newDataFlatArray.vectorFloat();
+		for (int i = 0; i < floatTargetDataVector->size(); i++) {
+			floatTargetDataVector->replace(i, floatTargetDataVector->value(i) + floatSourceDataVector->value(i));
+		}
+		break;
+	}
+	case AMDSDataTypeDefinitions::Double: {
+		QVector<double> *doubleSourceDataVector = &dataFlatArray.vectorDouble();
+		QVector<double> *doubleTargetDataVector = &newDataFlatArray.vectorDouble();
+		for (int i = 0; i < doubleTargetDataVector->size(); i++) {
+			doubleTargetDataVector->replace(i, doubleTargetDataVector->value(i) + doubleSourceDataVector->value(i));
+		}
+		break;
+	}
+	case AMDSDataTypeDefinitions::InvalidType:
+	default:
+		break;
+	}
+
+	return newDataFlatArray;
+}
+
+AMDSFlatArray AMDSFlatArray::operator /(const quint32 divisor)
```

Comment:
I would then argue that you may want to use a different divisor then.  By using an int, are there cases where you could be performing integer division?  If so, I'd recommend switching to using a floating point divisor to guarantee you get the division you expect.